File: andorid/app/src/main/java/com/gaudiyakirtan/myapplication/models/Topic.kt
package com.gaudiyakirtan.myapplication.models
import java.util.UUID

data class Topic(
    val id: UUID = UUID.randomUUID(),
    val name: String
) {
    // Generate a pseudo-random count based on the topic name length for demo purposes
    val demoSongCount: Int
        get() = (name.length * 3) % 20 + 1
}

File: web/src/models/Topic.ts
export interface ITopic {
  name: string;
}

export class Topic implements ITopic {
  name: string;
  
  constructor(name: string) {
    this.name = name;
  }
  
  // Generate a pseudo-random count based on the topic name length for demo purposes
  get demoSongCount(): number {
    return (this.name.length * 3) % 20 + 1;
  }
}